domain/resource/types.go

```diff
@@ -14,6 +14,19 @@ import (
 	charmresource "github.com/juju/juju/internal/charm/resource"
 )
 
+// StateType indicates if a resource is downloaded onto the controller
+// (available), or known to be available on charmhub (potential).
+type StateType string
+
+func (s StateType) String() string {
+	return string(s)
+}
+
+const (
+	StatePotential StateType = "potential"
+	StateAvailable StateType = "available"
+)
```

Comment:
Comment me, so we know when to use one vs the other.

Reply:
Commented, and clarified the one above.